### PR TITLE
operator/network: add standard OperatorStatus/Spec members

### PR DESCRIPTION
**Pull request by squeed**
This adds the standard OperatorStatus and OperatorSpec members to the struct, allowing the CNO to act a bit more like a "normal" operator.

### CODE REVIEW

File: operator/v1/types_network.go [hunk 1]
```diff
@@ -35,6 +36,8 @@ type NetworkList struct {
 
 // NetworkSpec is the top-level network configuration object.
 type NetworkSpec struct {
+	OperatorSpec `json:",inline"`
```
Comment (danwinship) [inline operator/v1/types_network.go:39]: You should remove the existing `LogLevel` field now so there's no ambiguity between the `NetworkSpec` and `OperatorSpec` ones

**Review by danwinship — COMMENTED** [review]
Are we actually planning to implement all of those fields? Should we have a CNO PR ready to go before merging this?

File: operator/v1/types_network.go (hunk 1 above)
Comment (squeed) [reply operator/v1/types_network.go:39]: Oh, good catch.

**Comment by squeed** [comment]
> Are we actually planning to implement all of those fields? 

We get most of them for free when we switch to the library-go tooling. The biggest open question would be ObservedConfig, which we should use for things like MTU that we infer unless overriden. But we don't need to use that just yet.

> Should we have a CNO PR ready to go before merging this?

Generally yes, but I expect it to be a few PRs just for logistical purposes (unblocking people). I can hold this until the first one is filed.

**Comment by squeed** [comment]
Rebased, and I have a PR implementing this. This is ready for final lgtm.

**Pull request merged**